### PR TITLE
implemented server side pagination

### PR DESCRIPTION
**Pull request by OwaisRaza**

### CODE REVIEW

**Comment by DaniyalKukda** [comment]
I think we don't need to do server side pagination for customers and companies because we have limited customers and companies antd table default handle the pagination for small data we don't to request an api call


cc: @hammadtariq

**Comment by hammadtariq** [comment]
> @hammadtariq I think we don't need to do server side pagination for customers and companies because we have limited customers and companies antd table default handle the pagination for small data we don't need to request an api call

lets keep it this way, because i am not bounding features according to my needs, there might be a user which have many companies or customers

File: pages/customers/index.js [hunk 1]
```diff
@@ -28,6 +28,11 @@ export default function Customers() {
     action: "edit",
   });
 
+  const paginationHandler = (limit, offset) => {
+    setLimit(limit);
+    setOffset(offset);
+  };
+
```
Comment (hammadtariq) [inline pages/customers/index.js:35]: we can move this to hooks file, and pass it directly to App table

File: hooks/company.js [hunk 2]
```diff
@@ -1,13 +1,20 @@
+import { useState } from "react";
 import useSWR from "swr";
 import { get, post, put, remove } from "@/lib/http-client";
 
 export const useCompanies = () => {
-  const { data, error, mutate } = useSWR("/api/company", get);
+  const [limit, setLimit] = useState(10);
+  const [offset, setOffset] = useState(0);
+  const { data, error, mutate } = useSWR(`/api/company?limit=${limit}&offset=${offset}`, get);
```
Comment (hammadtariq) [inline hooks/company.js:9]: `const paginationHandler = useCallback((limit, offset) => {
    setLimit(limit);
    setOffset(offset);
  }, [limit,offset]);`

File: hooks/company.js [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
 
   return {
     companies: data,
     isLoading: !error && !data,
     error,
+    limit,
+    offset,
+    setLimit,
+    setOffset,
```
Comment (hammadtariq) [inline hooks/company.js:17]:
```suggestion
paginationHandler
```

Comment (OwaisRaza) [reply hooks/company.js]: implemented.

**Comment by OwaisRaza** [comment]
implemented all changes, please have a look.

**Pull request merged**